images/install-s3-service

```diff
@@ -0,0 +1,60 @@
+#!/bin/sh
+
+# deploy this script with:
+# ansible -i inventory -m include_role -a name=s3-systemd e2e_s3
+set -eufx
+
+CACHE=/var/cache/cockpit-tasks
+SECRETS=/var/lib/cockpit-secrets
+
+if RUNC=$(which podman 2>/dev/null); then
+    UNIT_DEPS=''
+else
+    RUNC=$(which docker)
+    UNIT_DEPS="Requires=docker.service
+After=docker.service"
+fi
+
+systemctl stop cockpit-s3.service || true
+
+dd if=/dev/urandom bs=12 count=1 status=none | base64 > /run/cockpit-s3.rootpassword.txt
+
+cat <<EOF > /etc/systemd/system/cockpit-s3.service
+[Unit]
+Description=Cockpit S3 image server
+$UNIT_DEPS
+
+[Service]
+Restart=always
+RestartSec=60
+ExecStartPre=-$RUNC rm -f cockpit-s3
+ExecStartPre=/bin/rm -rf $CACHE/.minio.sys/
+ExecStart=/bin/sh -xc "$RUNC run --name=cockpit-s3 --publish=443:9000 -e MINIO_ROOT_USER=minioadmin -e MINIO_ROOT_PASSWORD=\$(cat /run/cockpit-s3.rootpassword.txt) -v $SECRETS/tasks/server.key:/root/.minio/certs/private.key:ro -v $SECRETS/tasks/server.pem:/root/.minio/certs/public.crt:ro -v $CACHE:/data:rw quay.io/minio/minio server /data"
+ExecStartPost=/usr/local/lib/setup-s3.sh
+ExecStop=$RUNC rm -f cockpit-s3
+
+[Install]
+WantedBy=multi-user.target
+EOF
+
+cat <<EOF > /usr/local/lib/setup-s3.sh
+#!/bin/sh
+set -eu
+read s3user s3key < "$SECRETS/tasks/s3-keys/e2e.bos.redhat.com"
+$RUNC run --interactive --rm --network=host \
+    -v "$SECRETS"/webhook/ca.pem:/etc/pki/ca-trust/source/anchors/ca.pem:ro \
+    --entrypoint /bin/sh quay.io/minio/mc <<EOC
+set -e
+update-ca-trust
+until mc alias set minio https://localhost minioadmin \$(cat /run/cockpit-s3.rootpassword.txt); do sleep 1; done
+# this fails with non-zero as the directory already exists; that's fine
+mc mb minio/images || true
+mc policy set download minio/images
+mc admin user add minio/ '\$s3user' '\$s3key'
+mc admin policy set minio/ readwrite user='\$s3user'
+EOC
```

Comment:
I think this is the first time I've seen this.

Reply:
Like, how to configure S3 user policies? Me too, if there is a more elegant way, I'm all ears. I was following https://docs.min.io/minio/baremetal/reference/minio-mc-admin/mc-admin-user.html

Reply:
I mean the double-nested here-document :)